### PR TITLE
convert_id script added 

### LINKED ISSUE #6: CISC ID to behavioural data ID conversion
Need a converter and encrypt the dictionary for conversion.

### PR DESCRIPTION
**Pull request by willstrawson**
Here's the first draft of this script that converts CISC IDs in a BIDS dataset to ADIE IDs. Running into an error on line 62, and need to make paths relative to user of script - will take another look after a brain break.

### CODE REVIEW

File: bin/cisc_to_adie_conversion.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+with open(txt) as f:
+    for line in f:
+        (key,val)=line.split()
+        rename[str(key)] = str(val)
+
+
+# USE OS.WALK
+
+def subconvert(p):
+# P should = the path of the directory above the sub- dirs
```
Comment (htwangtw) [inline bin/cisc_to_adie_conversion.py]: Look up the standard python function documentation format. Have a look at any function I wrote.

File: bin/cisc_to_adie_conversion.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+
+# USE OS.WALK
+
+def subconvert(p):
+# P should = the path of the directory above the sub- dirs 
+    for root,dirs,files in os.walk(p):
+        for f in files:
+            # Extract sub- number for searching 
+            # Extract number after 'sub-' and before '_'
```
Comment (htwangtw) [inline bin/cisc_to_adie_conversion.py]: We can improve the current search pattern to "sub-([0-9]*)". String the bracket will be recognised as a group.
I suggest using a dictionary so you are searching for the exact match.
Take the file name for example:
```python
# dictionary entry - CISCID : ADIEIDE 
# strip off CISC when saving to a dict
conversion = {"34233": "ADIE999", "54364": "ADIE998" .... }

cisc_bidsname = "sub-34233_stuff.nii.gz"
cur_subid = re.search("sub-([0-9]*)", cisc_bidsname).groups()[0]
adie_bidsname = cisc_bidsname.replace(cur_subid, conversion[cursubid])
```

File: bin/cisc_to_adie_conversion.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        (key,val)=line.split()
+        rename[str(key)] = str(val)
+
+
+# USE OS.WALK
+
+def subconvert(p):
+# P should = the path of the directory above the sub- dirs 
+    for root,dirs,files in os.walk(p):
+        for f in files:
```
Comment (htwangtw) [inline bin/cisc_to_adie_conversion.py:77]: Can you break your for loops and each step into smaller functions?
This way you can test them separately and know exactly what was wrong.
It's a bit difficult to work things out on just a line number and one test case.

File: bin/cisc_to_adie_conversion.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                srch = re.search('sub-(.+?)_',f)
+                # extract just ID number
+                cidn = srch.group(1)
+                # Add CISC to ID to enable search 
+                cid = 'CISC'+str(cidn)
+                # If this file contrains the CISC ID...
+                
+                if cid in rename.keys():
+                    #print (cid,rename[cid])
+                    try:
```
Comment (htwangtw) [inline bin/cisc_to_adie_conversion.py]: This should be in a function of its own

File: bin/cisc_to_adie_conversion.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+# USE OS.WALK
+
+def subconvert(p):
+# P should = the path of the directory above the sub- dirs 
+    for root,dirs,files in os.walk(p):
+        for f in files:
+            # Extract sub- number for searching 
+            # Extract number after 'sub-' and before '_'
+            try:
```
Comment (htwangtw) [inline bin/cisc_to_adie_conversion.py]: This should be a function of its own

**Review by htwangtw — CHANGES_REQUESTED** [review]
I am going to change the way I help people a bit due to now I have quite a lot on my plate. I will not be able to run your code so I won't unless necessary. 

If you can set up your github action with the existing configuration, the github action will pick it up some debugging. This only apply to function and tests stored under `adie` and the test folder. So you will have `adie/cisc.py` and `adie/tests/test_cisc.py`. Please see how I write test cases with the existing files.
With the developer requirements installed, you can run a test locally before pushing a commit to the remote branch.
Like this: `pytest adie/tests/test_cisc.py` 

If you are not comfortable with that, I still need you to break down your code into smaller functions. I only one for loop in the main script, and that is for looping through subject ID. Everything else should be their own function. Based on your current solution, I imagine it will be something like this: 
```
# for each subject
    # get all file path of this subject
    # create new dir with ADIE id of the same structure
    # move and rename file
```
Each comment is a function. Of course, how that's implemented is your choice.
See what I did in `bin/nback_derivative.py ` for example (I looped through files, but in the case of ID conversion subject makes more sense).

File: bin/cisc_to_adie_conversion.py (hunk 3 above)
Comment (willstrawson) [reply bin/cisc_to_adie_conversion.py:77]: Yeah these are all really good suggestions, thanks.

**Comment by htwangtw** [comment]
Close this PR due to duplication of #16

**Pull request closed without merging**